docs/book/v1/intro.md

```diff
@@ -112,6 +126,8 @@ grants are configured to be available. If you would like to disable any of the
 supplied grants, change the value for the grant to `null`. Additionally,
 you can extend this array to add your own custom grants.
 
+The `listeners` array is for enabling event listeners. Listeners are not required. This is an array of arrays. Each array in the list must contain at least 2 elements. The first element must be a string that corresponds to the name of the event to listen for. The second element must be either a string or an anonymous function. If it's a string it's assumed to be a container service key that points to your listener. There may be a third element of the array -- it must be an integer. The third element is the `$priority` argument when registering the listener. See the [Authorization Server Domain Events documentation](https://oauth2.thephpleague.com/authorization-server/events/).
```

Comment:
I'm missing the events here. There are only 2 listed in the league oauth2 docs. I think we need a link to this https://github.com/thephpleague/oauth2-server/blob/master/src/RequestEvent.php#L17-L22, or maybe describe each event here.